### ISSUE TITLE
Build order issue with default make command

### ISSUE THREAD

**Issue by jetvova** [pulumi/pulumi#7696]
CONTRIBUTING.md suggests to use `make` to build Pulumi and run a series of tests. However, it seems to build the project in an incorrect order which leads to some tests failing. 

## Steps to reproduce

Running `make` in the Pulumi folder leads to the Pulumi dotnet sdk being being built first. `LocalPulumiCmdTests.cs` then attempt to run the Pulumi command line executable, which hasn't been built yet leading to a `System.ComponentModel.Win32Exception : No such file or directory` error.

Running `make build` creates the necessary `pulumi` file, after which `make` is able to run the tests properly.

**Comment by iwahbe** [comment]
I can confirm that this works.

**Comment by pkit** [comment]
> Running make build creates the necessary pulumi file, after which make is able to run the tests properly.

Nope. Still fails with `stderr: error: PULUMI_ACCESS_TOKEN must be set for login during non-interactive CLI sessions`

**Comment by lunaris** [comment]
There's definitely a way for us to go to make contributing easier here, but as far as I know, running `make build` in the repository today should work just fine without any Pulumi access tokens, etc. To this end I'm going to close this, but we can revisit if I'm wrong (quite possible!) 🙏
